src/main/java/eu/xenit/alfred/content/gateway/servicediscovery/DisabledServiceDiscovery.java

```diff
@@ -0,0 +1,10 @@
+package eu.xenit.alfred.content.gateway.servicediscovery;
+
+
+public class DisabledServiceDiscovery implements ServiceDiscovery {
```

Comment:
Instead of a `DisabledServiceDiscovery` class that does nothing, couldn't you just not create an `ApplicationRunner` when no `ServiceDiscovery` bean is available? Or even make `KubernetesServiceDiscovery` implement `ApplicationRunner`, so it's just run automatically if the bean is registered.

Reply:
Yeah I'm doing that as part of moving the whole service discovery stuff to its own `@Configuration`